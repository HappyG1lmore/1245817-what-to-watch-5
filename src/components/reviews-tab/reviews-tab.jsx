import React from "react";
import {connect} from "react-redux";
import {REVIEWS_COLUMN_COUNT} from "../../constants";
import moment from "moment";
import PropTypes from "prop-types";

const ReviewsTab = (props) => {

  const {comments} = props;
  const commentsList = [comments.slice(0, REVIEWS_COLUMN_COUNT), comments.slice(REVIEWS_COLUMN_COUNT, REVIEWS_COLUMN_COUNT * 2)];

  return (
    <div className="movie-card__reviews movie-card__row">
      {commentsList && commentsList.map((reviewsColumn, idx) => (
        <div
          className="movie-card__reviews-col"
          key={idx}
        >
          {reviewsColumn.map((review) => (
            <div className="review" key={review.id}>
              <blockquote className="review__quote">
                <p className="review__text">{review.comment}</p>
                <footer className="review__details">
                  <cite className="review__author">{review.user.name}</cite>
                  <time
                    className="review__date"
                    dateTime={`${moment(review.date).format(`YYYY-MMMM-DD`)}`}
                  >
                    {moment(review.date).format(`MMMM DD, YYYY`)}
                  </time>
                </footer>
              </blockquote>

              <div className="review__rating">
                {String(review.rating).replace(`.`, `,`)}
              </div>
            </div>
          ))}
        </div>
      ))}
    </div>
  );
};

ReviewsTab.propTypes = {
  comments: PropTypes.arrayOf(
      PropTypes.shape({
        comment: PropTypes.string,
        date: PropTypes.string,
        id: PropTypes.number,
        rating: PropTypes.number,
        user: PropTypes.shape({
          id: PropTypes.number,
          name: PropTypes.string,
        })
      }))
};

const mapStateToProps = (state) => {
  return {
    comments: state.reviews.comments
  };
};

export {ReviewsTab};
export default connect(mapStateToProps)(ReviewsTab);
