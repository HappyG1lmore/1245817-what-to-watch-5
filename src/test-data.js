const filmsRaw = [{
  "name": `Bronson`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  "background_color": `#73B39A`,
  "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  "rating": 3.6,
  "scores_count": 109661,
  "director": `Nicolas Winding Refn`,
  "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  "run_time": 92,
  "genre": `Action`,
  "released": `2008`,
  "id": 1,
  "is_favorite": false,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "name": `Pulp Fiction`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
  "background_color": `#795433`,
  "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
  "rating": 1.5,
  "scores_count": 1635992,
  "director": `Quentin Tarantino`,
  "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
  "run_time": 153,
  "genre": `Crime`,
  "released": `1994`,
  "id": 2,
  "is_favorite": false,
  "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "name": `Matrix`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
  "background_color": `#B9B27E`,
  "description": `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
  "rating": 4.4,
  "scores_count": 1503092,
  "director": `Wachowski Brothers`,
  "starring": [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
  "run_time": 136,
  "genre": `Action`,
  "released": `1999`,
  "id": 4,
  "is_favorite": false,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}];

const films = [{
  "title": `Bronson`,
  "poster": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  "frame": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  "background": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  "backgroundColor": `#73B39A`,
  "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  "ratingText": `Normal`,
  "rating": 3.6,
  "ratings": `109661`,
  "director": `Nicolas Winding Refn`,
  "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  "runtime": 92,
  "genre": `Action`,
  "year": `2008`,
  "id": 1,
  "isFavorite": false,
  "video": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "previewVideoLink": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "title": `Pulp Fiction`,
  "poster": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
  "frame": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
  "background": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
  "backgroundColor": `#795433`,
  "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
  "ratingText": `Bad`,
  "rating": 1.5,
  "ratings": `1635992`,
  "director": `Quentin Tarantino`,
  "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
  "runtime": 153,
  "genre": `Crime`,
  "year": `1994`,
  "id": 2,
  "isFavorite": false,
  "video": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  "previewVideoLink": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "title": `Matrix`,
  "poster": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
  "frame": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
  "background": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
  "backgroundColor": `#B9B27E`,
  "description": `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
  "ratingText": `Normal`,
  "rating": 4.4,
  "ratings": `1503092`,
  "director": `Wachowski Brothers`,
  "starring": [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
  "runtime": 136,
  "genre": `Action`,
  "year": `1999`,
  "id": 4,
  "isFavorite": false,
  "video": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "previewVideoLink": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}];

const film = {
  "title": `Pulp Fiction`,
  "poster": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
  "frame": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
  "background": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
  "backgroundColor": `#795433`,
  "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
  "rating": 1.5,
  "ratings": `1635992`,
  "director": `Quentin Tarantino`,
  "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
  "runtime": 153,
  "genre": `Crime`,
  "year": `1994`,
  "id": 2,
  "isFavorite": false,
  "video": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  "previewVideoLink": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};

const reviews = [{
  "id": 1,
  "user": {
    "id": 19,
    "name": `Christina`
  },
  "rating": 3.3,
  "comment": `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
  "date": `2020-10-09T13:38:44.769Z`
}, {
  "id": 2,
  "user": {
    "id": 25,
    "name": `Sophie`
  },
  "rating": 9,
  "comment": `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
  "date": `2020-11-04T13:38:44.769Z`
}, {
  "id": 3,
  "user": {
    "id": 7,
    "name": `Jack`
  },
  "rating": 9,
  "comment": `Unfortunately we don't have a reliable way to tell the true ratings of a movie.`,
  "date": `2020-10-09T13:38:44.769Z`
}];

const avatarUrl = `/avatar.jpg`;

const user = {
  "id": 1,
  "email": `Oliver.conner@gmail.com`,
  "name": `Oliver.conner`,
  "avatar_url": `img/1.png`
};

const noop = () => {};

const genres = [`Comedy`, `Drama`, `Thriller`];

export {films, genres, reviews, avatarUrl, user, film, filmsRaw, noop};
