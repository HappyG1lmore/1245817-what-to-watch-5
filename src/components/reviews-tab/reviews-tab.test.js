import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {ReviewsTab} from './reviews-tab';
import {reviews} from '../../test-data';

const renderer = new ShallowRenderer();

describe(`ReviewsTab renders correctly`, () => {
  it(`When film has reviews`, () => {
    renderer.render(<ReviewsTab comments={reviews}/>);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it(`When film has no reviews`, () => {
    renderer.render(<ReviewsTab comments={[]}/>);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
