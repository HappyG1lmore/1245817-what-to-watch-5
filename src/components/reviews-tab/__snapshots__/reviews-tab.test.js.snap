// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsTab renders correctly When film has no reviews 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  />
  <div
    className="movie-card__reviews-col"
  />
</div>
`;

exports[`ReviewsTab renders correctly When film has reviews 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Christina
          </cite>
          <time
            className="review__date"
            dateTime="2020-October-09"
          >
            October 09, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3,3
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Sophie
          </cite>
          <time
            className="review__date"
            dateTime="2020-November-04"
          >
            November 04, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        9
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Unfortunately we don't have a reliable way to tell the true ratings of a movie.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Jack
          </cite>
          <time
            className="review__date"
            dateTime="2020-October-09"
          >
            October 09, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        9
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  />
</div>
`;
